### ISSUE TITLE
thread_count = -1 does not work with set_params()

### ISSUE THREAD

**Issue by MatteoFelici** [catboost/catboost#1800]
**Problem:**
If I set the parameter `thread_count` equal to -1 with the `set_params` method, and I try to fit the model

```
from catboost import CatBoostClassifier
import numpy as np

model = CatBoostClassifier()
model.set_params(thread_count=-1)

x = np.random.random((3, 2))
y = [1,0,0]

model.fit(x, y)
``` 

I get

```
_catboost.CatBoostError: catboost/private/libs/options/json_helper.h:171: Can't parse parameter "thread_count" with value: -1
```

**catboost version:** 0.26
**Operating System:** Mac OS Big Sur 11.2.1

**Comment by grag90** [comment]
Thank you for reporting bug, it will be fixed in ours next release.

**Comment by kizill** [comment]
Released in 1.0.0
